packages/core/src/anchor/memory/in-memory-anchor-service.ts

```diff
@@ -287,34 +310,51 @@ export class InMemoryAnchorService implements AnchorService, AnchorValidator {
     }
   }
 
+  async _storeRecord(record: Record<string, unknown>): Promise<CID> {
+    let timeout: any
+
+    const putPromise = this.#dispatcher.storeCommit(record).finally(() => {
+      clearTimeout(timeout)
+    })
+
+    const timeoutPromise = new Promise((resolve) => {
+      timeout = setTimeout(resolve, 30 * 1000)
+    })
+
+    return await Promise.race([
+      putPromise,
+      timeoutPromise.then(() => {
+        throw new Error(`Timed out storing record in IPFS`)
+      }),
+    ])
+  }
+
   /**
    * Sends anchor commit to Ceramic node and instructs it to publish the commit to pubsub
    * @param streamId
    * @param commit
    */
   async _publishAnchorCommit(streamId: StreamID, commit: AnchorCommit): Promise<CID> {
-    const block = await Block.encode({ value: commit, codec, hasher })
-    const expectedCID = block.cid
-    const stream = await this.#ceramic.applyCommit(streamId, commit, {
-      publish: true,
-      anchor: false,
+    const anchorCid = await this._storeRecord(commit as any)
+    let resolved = false
+    await new Promise<void>((resolve) => {
```

Comment:
why do you need this wrapping Promise?  Can you not just `await this.#dispatcher.publishTip()`?

Reply:
`publishTip()` returns an instance of `Subscription` and they are not awaitable as far as I understand :)

Reply:
ah okay, do we even need to wait for it at all then?  The pubsub message is processed asynchronously anyway, so I'm not really sure what value we're getting from waiting here?